trpcultivate_germcollection/src/Plugin/TripalImporter/GermplasmRelationshipImporter.php

```diff
@@ -0,0 +1,1193 @@
+              }
+            }
+            else {
+              // Relationship only:
+              // Fetch the stock_id of the individual using
+              // the uniquename provided in the file.
+              $individual = NULL;
+              $query = $this->chado_connection->select('1:stock', 's')
+                ->fields('s', ['stock_id'])
+                ->condition('s.name', $val_name, '=')
+                ->condition('s.type_id', $type_id, '=')
+                ->condition('s.organism_id', $organism_id, '=')
+                ->execute();
+
+              if ($stock_id = $query->fetchField()) {
+                $individual = $stock_id;
+              }
+            }
+
+            // Create Relationship:
+            $relation = [];
+            // Verb.
+            $relation['type_id'] = ChadoCVTermAutocompleteController::getCVtermId($population['verb']);
+
+            // Position.
+            if ($population['position'] == 'evi') {
+              // Entry - Verb - Individual.
+              $relation['subject_id'] = $population['entry'];
+              $relation['object_id'] = $individual;
+            }
+            elseif ($population['position'] == 'ive') {
+              // Individual - Verb - Entry.
+              $relation['object_id'] = $population['entry'];
+              $relation['subject_id'] = $individual;
+            }
+
+            $this->chado_connection->insert('1:stock_relationship')
+              ->fields($relation)
+              ->execute();
+            unset($individual);
+
+            $i++;
+          }
+        }
+
+        // Close file.
+        fclose($handle);
+      }
+    }
+  }
+
+  /**
+   * Parse form values for Stock/Germplasm (Primary Germplasm).
+   *
+   * Fetch the matching row in chado.stock table.
+   *
+   * @param string $stock_name
+   *   String, containing the stock name and  in
+   *   the following notation: Stock Name.
+   * @param int $type_id
+   *   Integer, containing the cvterm type id.
+   * @param int $organism_id
+   *   Integer, containing organism_id of the scientific name.
+   *
+   * @return int
+   *   Stock id number that matched the resolved stock id
+   *   from the input string.
+   */
+  public function parseStock($stock_name, $type_id, $organism_id) {
+    $result = '';
+
+    // Fetch the germplasm name and return
+    // the stock_id number.
+    $query = $this->chado_connection->select('1:stock', 's')
+      ->fields('s', ['stock_id'])
+      ->condition('s.name', $stock_name, '=')
+      ->condition('s.type_id', $type_id, '=')
+      ->condition('s.organism_id', $organism_id, '=')
+      ->execute();
+
+    $result = NULL;
+    if ($stock_id = $query->fetchField()) {
+      $result = $stock_id;
+    }
+
+    return $result;
+  }
+
+  /**
+   * Get organism ids, or use the previously looked up organisms.
+   *
+   * @param string $organism
+   *   The scientific name of the organism.
+   */
+  public function getOrganismIds($organism) {
```

Comment:
Type hint the param and return value. Also document the return value which I think should be an integer (0 if no organism is found).